utils/storage/storage.js

```diff
@@ -248,13 +251,84 @@ class Storage {
     );
   }
 
+  /**
+   * Function to fetch anonymousId from external source
+   * @param {string} key source of the anonymousId
+   * @returns string
+   */
+  fetchExternalAnonymousId(source) {
+    let anonId;
+    const key = source.toLowerCase();
+    if (!Object.keys(anonymousIdKeyMap).includes(key)) {
+      return anonId;
+    }
+    switch (key) {
+      case "segment":
+        /**
+         * First check the local storage for anonymousId
+         * Ref: https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#identify
+         */
+        if (Store.enabled) {
+          anonId = Store.get(anonymousIdKeyMap[key]);
+        }
+        // If anonymousId is not present in local storage and check cookie support exists
+        // fetch it from cookie
+        if (!anonId && Cookie.IsCookieSupported()) {
+          anonId = Cookie.get(anonymousIdKeyMap[key]);
+        }
+        return anonId;
+
+      default:
+        return anonId;
+    }
+  }
+
   /**
    * get stored anonymous id
+   *
+   * Use cases:
+   * 1. getAnonymousId() ->  anonymousIdOptions is undefined this function will return rl_anonymous_id
+   * if present otherwise undefined
+   *
+   * 2. getAnonymousId(anonymousIdOptions) -> In case anonymousIdOptions is present this function will check
+   * if rl_anonymous_id is present then it will return that
+   *
+   * otherwise it will validate the anonymousIdOptions and try to fetch the anonymous Id from the provided source.
+   * Finally if no anonymous Id is present in the source it will return undefined.
+   *
+   * anonymousIdOptions example:
+   *  {
+        autoCapture: {
+          enabled: true,
+          source: "segment",
+        },
+      }
+   *
    */
-  getAnonymousId() {
-    return this.parse(
+  getAnonymousId(anonymousIdOptions) {
+    // fetch the rl_anonymous_id from storage
+    const rlAnonymousId = this.parse(
       this.decryptValue(this.storage.get(defaults.user_storage_anonymousId))
     );
+    /**
+     * If rl_anonymous_id is already present return the value rather then executing the rest of the block
+     */
```

Comment:
```suggestion
    // If RS's anonymous ID is already present, return from here.
    // The user, while migrating from a different analytics SDK, will only need to auto-capture the anonymous ID when the RS SDK loads for the first time. 
    // The captured anonymous ID would be available in RS's persistent storage for all the subsequent SDK runs. 
    // So, instead of always grabbing the ID from the migration source when the options are specified, it is first checked in the RS's persistent storage.
    // Moreover, the user can also clear the anonymous ID from the storage via the 'reset' API, which renders the migration source's data useless.
    ```